File: java/arrays/How_Many_Numbers_Are_Smaller_Than_the_Current_Number_1365.java
package java.arrays;

public class How_Many_Numbers_Are_Smaller_Than_the_Current_Number_1365 {
        public int[] smallerNumbersThanCurrent(int[] nums) {
    //  int[] res=new int[nums.length];
    //  for(int i=0;i<nums.length;i++){
    //    int curr_sum=0;
    //     for(int j=0;j<nums.length;j++){
    //         if (nums[i]>nums[j]){
    //             curr_sum++;
    //         }
    //     }
    //     res[i]=curr_sum;
    //  } 
    //  return res;  


// or

    int max=101;
    int[] freq_cnt=new int[max];
    for(int num : nums){
        freq_cnt[num]+=1;

    }
    for(int i=1;i<max;i++){
        freq_cnt[i]+=freq_cnt[i-1];
    }
    int[] res=new int[nums.length];
    for (int i=0;i<nums.length;i++){
        if (nums[i]==0){
            res[i]=0;
        }
        else{
            res[i]=freq_cnt[nums[i]-1];
        }
    }
    return res;
    }
}

File: python/arrays/How_Many_Numbers_Are_Smaller_Than_the_Current_Number_1365.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        # res=[0]*len(nums)
        # for i in range(len(nums)):
        #     curr_count=0
        #     for j in range(len(nums)):
        #         if nums[i]>nums[j]:
        #             curr_count+=1
        #     res[i]=curr_count
        # return res

# or

        max_val=101
        count_freq=[0]*max_val
        for num in nums:
            count_freq[num]+=1
        for i in range(1,max_val):
            count_freq[i]+=count_freq[i-1]
        res=[]
        for num in nums:
            if num==0:
                res.append(0)
            else:
                res.append(count_freq[num-1])
        return res